### PR TITLE
script now pushes to google spreadsheet

### PR DESCRIPTION
**Pull request by HassenPy**
summary of what's been added:

- JSON as a db to hold all events, a spreadsheet felt like reinventing a db and a "real" db engine is too much.
- Events that are fetched from a fork are now stored to a central json file (preserved the file saving logic).
- Writing events to the google spreadsheet.
- comments :laughing: 

tried to preserve the coding style as much as possible, i ll be happy to make this in es6 syntax, will be easier to read and change ...

to test this you need turn on the Google Sheets API, step 1 of this guide:
https://developers.google.com/sheets/api/quickstart/nodejs

and specify the sheet id you want to write to in an env variable SHEET_ID.

this is an example result sheet:
https://docs.google.com/spreadsheets/d/1HiidDypELA_D6CRbKJL4rYKoiQ2cHvht54p0TRUt_QU/

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -102,35 +241,225 @@ function processForkBatch(user, repo, offset, size){
...
-                console.log(file.path);
-                //use bluebird spread
-                promises.push(repo.contents(file.path).read());
-            });
+      var promises = [];
+      contents.map(function(file){
+        promises.push(
+          // Needed to add an explicit .catch on this so that it returns,
+          // else it won't be possible for the promise.all() to resolve...
+          repo.contents(file.path).read().catch(function(e){
```
Comment (HassenPy) [inline index.js]: this is one of the devils that took me some time to find.

File: index.js [hunk 2]
```diff
@@ -102,35 +241,225 @@ function processForkBatch(user, repo, offset, size){
...
+        console.error('Batch create file error');
+        console.error('ERROR: %s.\n%s', err.message, err.stack);
+        return reject('Batch create file error');
+      });
+    }).catch(function(err){
+      console.error('Fetch repo _posts error');
+      console.error('ERROR: %s.\n%s', err.message, err.stack);
+      return reject('Fetch repo _posts error');
     });
+  }).catch(function(e){
```
Comment (HassenPy) [inline index.js]: and this is the other one

**Comment by HassenPy** [comment]
i ll work on the studyGroupEvents repo once this gets approved.

**Comment by AnamikaD** [comment]
@HassenPy Amazing Job! I would try to run that script. With the help of your sheet, we can present the data on [Study Group Events](https://mozillascience.github.io/studyGroupEvents/)!
@auremoser Look at this !! :D

File: index.js (hunk 1 above)
Comment (auremoser) [inline index.js]: :) good work!

**Comment by auremoser** [comment]
This is excellent @HassenPy , thank you SO MUCH for these modifications, all checks out and passes. I'm so sorry about the delays!

**Comment by AnamikaD** [comment]
@HassenPy Congratulation! Good Job :D

**Comment by HassenPy** [comment]
thanks everyone!

**Pull request merged**